staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -237,7 +237,7 @@ type StatefulSetStatus struct {
 	// replicas is the number of Pods created by the StatefulSet controller.
 	Replicas int32 `json:"replicas" protobuf:"varint,2,opt,name=replicas"`
 
-	// readyReplicas is the number of Pods created by the StatefulSet controller that have a Ready Condition.
+	// Total number of pods created by this StatefulSet controller with a Ready Condition by passing the readinessProbe.
```

Comment:
comment should start with the serialized field name (`readyReplicas`)

applies to all modified godoc lines

Reply:
Does this accurately describe behavior for pods without readiness probes? What about pods that set spec.readinessGates?

Referencing the Ready condition makes sense... I'm not sure denormalizing all the inputs that go into determining the Ready condition here makes sense (applies to all modified godoc lines)

Reply:
> Does this accurately describe behavior for pods without readiness probes? What about pods that set spec.readinessGates?

Good point. I rechecked the code and in the file podutils.go the function IsPodReady basically  just checks the Ready Condition. In the pkg/kubelet/status/generate.go file is the function GeneratePodReadyCondition defined, which checks, that every container has the status ready and all readinessGates are true.

According to the KEP documentation https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/580-pod-readiness-gates is ready defined as "ready condition and all readiness gates are true". 

My suggestion would be to have a text like this:
```suggestion
	// readyReplicas is the number of pods created by this StatefulSet controller with a Ready Condition and that are passing all readinessGates.
```

Or would you suggest a rather generic description? Do you have another suggestion?

Reply:
I wouldn't try to describe everything that goes into determining the Ready condition here... just limit it to saying it's the count of pods created by this controller with a Ready condition

Reply:
👍
Next week I will commit a new version.